tests/unit/test_results.py

```diff
@@ -0,0 +1,21 @@
+import earthaccess
+
+
+def test_data_links():
+    granules = earthaccess.search_data(
+        short_name="SEA_SURFACE_HEIGHT_ALT_GRIDS_L4_2SATS_5DAY_6THDEG_V_JPL2205",
+        temporal=("2020", "2022"),
+        count=1,
+    )
+    g = granules[0]
+    # `access` specified
+    assert g.data_links(access="direct")[0].startswith("s3://")
+    assert g.data_links(access="external")[0].startswith("https://")
+    # `in_region` specified
+    assert g.data_links(in_region=True)[0].startswith("s3://")
+    assert g.data_links(in_region=False)[0].startswith("https://")
+    # When `access` and `in_region` are both specified, `access` takes priority
+    assert g.data_links(access="direct", in_region=True)[0].startswith("s3://")
+    assert g.data_links(access="direct", in_region=False)[0].startswith("s3://")
+    assert g.data_links(access="external", in_region=True)[0].startswith("https://")
+    assert g.data_links(access="external", in_region=False)[0].startswith("https://")
```

Comment:
I think this is the intended behavior we want, but let me know if I'm missing something. 

As a side note, I'm not sure why we have separate `access` and `in_region` kwargs for determining if we want to use `s3` or `https` urls. Is one kwarg sufficient?

Reply:
>As a side note, I'm not sure why we have separate access and in_region kwargs for determining if we want to use s3 or https urls. Is one kwarg sufficient?

I can't answer the question directly, but these keywords also feel unintuitive to me. What about `access="s3"`? To me, "direct" and "external" don't mean anything without more context, but "s3" and "https" do.

Reply:
I agree that two kwargs is clunky and direct/external aren't the most descriptive names, and we could likely handle it with a single kwarg.

that said, since they are the current interface, we should probably open an issue for possibly refactoring it and not block this PR.